### DISCUSSION TITLE
how to use sortby and cumsum on lat-long time data?

### DISCUSSION THREAD

**Discussion by snowzky** [pydata/xarray#5403] [Q&A]
I am coming up short with xarrray. Maybe it can't do what I want, but I suspect my problem is me not understanding how to index/select dimensions/axes.

I have data that have dimensions lat, long and time. Just like the tutorial data. I would like to:
-  for each time, to get the flattened, sorted array of the map, ie. all the values in the lat-lon matrix defined by the specific time step, in order of lowest to highest (or highest to lowest)
- for each time, to get the cumsum flattened array of the map in similar fashion.

This I can get for one time step by using numpy's np.sort by selection time for the xarray dataarray.values. To do for all time steps, I need a loop.

Is there not functionality akin to how .sum works, where it - because it sums all elements - of course only return an dataarray with dimension and size of the time coordinate? If I just write the exact same, I get a returned array that still has a lat-lon matrix for each time, though it is cumsummed (by what axis order I do not know).

Regarding sortby I have tried many things too, but I don't know how to tell it to sort by the values of the lat-lon matrix for each time step.
The example in the docs sort by the array's own values (like I want) but it a 1D array with only time dimension.

What key thing am I missing here ...?
```import xarray as xr
ds = xr.tutorial.open_dataset("air_temperature.nc")
print(ds.air.sum(dim=['lat','lon']).shape)
>>> (2920,)
print(ds.air.cumsum(dim=['lat','lon']).shape)
>>> (2920, 25, 53)

**Comment by max-sixty** [comment]
Please could you add a copy-pasteable example?
